src/vm/eventpipebuffermanager.cpp

```diff
@@ -358,6 +347,9 @@ bool EventPipeBufferManager::WriteEvent(Thread *pThread, EventPipeSession &sessi
         }
         else
         {
+            // The event is still enabled.  Mark that the thread is now writing an event.
+            pThreadBufferList->SetThreadEventWriteInProgress(true);
+
             // Attempt to write the event to the buffer.  If this fails, we should allocate a new buffer.
             allocNewBuffer = !pBuffer->WriteEvent(pEventThread, session, event, payload, pActivityId, pRelatedActivityId, pStack);
```

Comment:
Should we be setting `pThreadBufferList->SetThreadEventWriteInProgress(false);` after each call to `WriteEvent` instead?

Reply:
It'll be set to ```false``` in line 385

Reply:
I feel it makes more sense to keep them together. It is less error prone.